packages/headless/src/controllers/commerce/facets/core/headless-core-commerce-facet.ts

```diff
@@ -125,14 +156,34 @@ export function buildCoreFacet(
   return {
     ...controller,
 
-    toggleSelect: (selection: FacetValue) =>
-      dispatch(toggleSelectFacetValue({facetId, selection})),
+    toggleSelect: (selection: AnyFacetValue) => {
+      if (props.options.toggleSelectActionCreator) {
```

Comment:
If I make the action creator props mandatory, that'll get rid of this awkwardness.